domain/agentbinary/state/state.go

```diff
@@ -74,7 +74,7 @@ WHERE  path = $objectStorePath.path`, objectStore)
 	return objectStore.MetadataUUID, nil
 }
 
-// Add adds a new agent binary's metadata to the database.
+// AddAgentBinary adds a new agent binary's metadata to the database.
```

Comment:
Would it be possible to add a quick note to `State` documenting the fact that this is currently written with the expectation that it can work over both a model and controller database? Just to future implementers understand the context and what we have done here?